packages/task/src/browser/quick-open-task.ts

```diff
@@ -234,77 +234,41 @@ export class QuickOpenTask implements QuickOpenModel, QuickOpenHandler {
         const providedTasks = await this.taskService.getProvidedTasks(token);
-
-        // check if tasks.json exists. If not, display "Create tasks.json file from template"
-        // If tasks.json exists and empty, display 'Open tasks.json file'
-        let isFirstGroup = true;
         const { filteredConfiguredTasks, filteredProvidedTasks } = this.getFilteredTasks([], configuredTasks, providedTasks);
         const groupedTasks = this.getGroupedTasksByWorkspaceFolder([...filteredConfiguredTasks, ...filteredProvidedTasks]);
-        if (groupedTasks.has(TaskScope.Global.toString())) {
-            const configs = groupedTasks.get(TaskScope.Global.toString())!;
-            this.items.push(
-                ...configs.map(taskConfig => {
-                    const item = new TaskConfigureQuickOpenItem(
-                        token,
-                        taskConfig,
-                        this.taskService,
-                        this.taskNameResolver,
-                        this.workspaceService,
-                        isMulti,
-                        { showBorder: false }
-                    );
-                    item['taskDefinitionRegistry'] = this.taskDefinitionRegistry;
-                    return item;
-                })
-            );
-            isFirstGroup = false;
+
+        const scopes: TaskConfigurationScope[] = [TaskScope.Global];
+        if (this.workspaceService.opened) {
+            scopes.push(TaskScope.Workspace);
+            if (this.workspaceService.isMultiRootWorkspaceOpened) {
+                const roots = await this.workspaceService.roots;
+                scopes.push(...roots.map(rootStat => rootStat.resource.toString()));
+            }
         }
 
-        const rootUris = (await this.workspaceService.roots).map(rootStat => rootStat.resource.toString());
-        for (const rootFolder of rootUris) {
-            const folderName = new URI(rootFolder).displayName;
-            if (groupedTasks.has(rootFolder)) {
-                const configs = groupedTasks.get(rootFolder.toString())!;
-                this.items.push(
-                    ...configs.map((taskConfig, index) => {
-                        const item = new TaskConfigureQuickOpenItem(
-                            token,
-                            taskConfig,
-                            this.taskService,
-                            this.taskNameResolver,
-                            this.workspaceService,
-                            isMulti,
-                            {
-                                groupLabel: index === 0 && isMulti ? folderName : '',
-                                showBorder: !isFirstGroup && index === 0
-                            }
-                        );
-                        item['taskDefinitionRegistry'] = this.taskDefinitionRegistry;
-                        return item;
-                    })
-                );
+        let isFirstGroup = true;
```

Comment:
I am wondering if we should replace this flag with this.items.length instead.

Reply:
I agree it's a bit unnecessary, but I think I prefer to keep the flag, because if someone later refactored to build the `items` array in the function scope and then assign it to `this.items` at the end, they could easily miss the side effect that that would have on the border display.

Reply:
It's ok to keep it then !, thanks !